File: app/models/address.rb
# == Schema Information
#
# Table name: addresses
#
#  id          :bigint           not null, primary key
#  address     :string           not null
#  address_2   :string
#  city        :string
#  country     :string
#  notes       :text
#  postal      :string
#  region      :string
#  created_at  :datetime         not null
#  updated_at  :datetime         not null
#  category_id :bigint           not null
#  contact_id  :bigint           not null
#
# Indexes
#
#  index_addresses_on_category_id  (category_id)
#  index_addresses_on_contact_id   (contact_id)
#
# Foreign Keys
#
#  fk_rails_...  (category_id => categories.id)
#  fk_rails_...  (contact_id => contacts.id)
#
class Address < ApplicationRecord
  include Categorizable

  tracked
  resourcify

  belongs_to :contact

  # def self.find_by_category(category)
  #   self.where(category:)
  # end
end

File: app/frontend/tests/helpers/fixtures/address.ts
const baseAddressCategory: Schema.CategoriesOptions = {
	id: 1,
	name: "Home",
	category_with_type: "Home (Address)",
	slug: "home",
}

export const baseAddressFixture: Schema.Address = {
	id: 1,
	address: "123 Main St",
	city: "New York",
	region: "NY",
	postal: "10001",
	country: "US",
	category_id: 1,
	contact_id: 1,
	category: baseAddressCategory,
}

export const createAddressFixture = (overrides?: Partial<Schema.Address>): Schema.Address => {
	return {
		...baseAddressFixture,
		...overrides,
		category: {
			...baseAddressCategory,
			...overrides?.category,
		},
	}
}

export const minimalAddressFixture: Schema.Address = {
	id: 1,
	address: "123 Main St",
	category_id: 1,
	contact_id: 1,
	category: baseAddressCategory,
}
